modules/gapi/src/streaming/engine/engine_session.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef GAPI_STREAMING_ENGINE_SESSION_HPP
+#define GAPI_STREAMING_ENGINE_SESSION_HPP
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#ifdef HAVE_ONEVPL
+#include <vpl/mfxvideo.h>
+
+namespace cv {
+namespace gapi {
+namespace wip {
+
+struct DecoderParams {
+    mfxBitstream stream;
+    mfxVideoParam param;
+};
+
+struct EngineSession {
+    mfxSession session;
+    mfxBitstream stream;
+    mfxSyncPoint sync;
```

Comment:
Is this some kind of mutex? Do we need any synchronization on our side?

Reply:
it handle used for VPL transaction synchronization
transaction: "Async Decode" -> "Sync Result". I suppose it seems something like `std::conditional_variable`

in the future it might associated with every pair `Decode<->Sync`. I mean  nor single  `sync` here but `queue<sync>`